sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/StateStore.scala

```diff
@@ -126,7 +126,10 @@ trait StateStore extends ReadStateStore {
 
   /**
    * Return an iterator containing all the key-value pairs in the StateStore. Implementations must
-   * ensure that updates (puts, removes) can be made while iterating over this iterator.
+   * ensure that updates (puts, removes) can be made while iterating over this iterator, but it is
```

Comment:
Split the ', but... ' part as a separate sentence. 

/**
   * Return an iterator containing all the key-value pairs in the StateStore. Implementations must
   * ensure that updates (puts, removes) can be made while iterating over this iterator.
   * It is not required for implementations ...

Reply:
Thanks for the suggestion! I'll reflect the change, and go merging since this is a nit comment.